### PR TITLE
Publish crate pagination docs, tests, and exec plan (4.1.3)

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Expand crate-level pagination documentation (ordering requirements, default/max limits, error mapping, and scope boundaries).
- Improve module-level documentation for `cursor`, `envelope`, and `params` modules to reflect integration details and edge-cases.
- Add a new ExecPlan document describing the plan for 4.1.3 (docs/execplans/backend-4-1-3-publish-crate-level-pagination-documentation.md).
- Update architecture and roadmap docs to reflect completion of item 4.1.3.
- Public API surface remains unchanged; changes are documentation and tests only.
- Add shared test fixtures and BDD driver support for documentation invariants in tests/common.rs.
- Update docs/developers-guide.md to describe shared workspace crate testing layout and guidance.

### Documentation
- Expand crate-level docs in `backend/crates/pagination/src/lib.rs` with four sections:
  - Ordering requirements
  - Default and maximum limits (documenting `DEFAULT_LIMIT = 20` and `MAX_LIMIT = 100`)
  - Error mapping guidelines for `CursorError` and `PageParamsError`
  - Scope boundaries (what the crate does not provide)
- Improve module-level documentation:
  - `cursor.rs` gains notes on base64url encoding, direction behavior, and cursor opacity, plus security considerations about opacity (not signed/encrypted).
  - `envelope.rs` notes behavior of `from_request` and optional links serialization, including `skip_serializing_if` behavior for nulls.
  - `params.rs` notes on `Deserialize` normalization and framework integration.
- New ExecPlan document: `docs/execplans/backend-4-1-3-publish-crate-level-pagination-documentation.md` describing the execution plan, scope, risks, and acceptance criteria.

### Tests
- Add unit tests (using `rstest`) to verify documented invariants in:
  - `params.rs` (consistency of `DEFAULT_LIMIT`/`MAX_LIMIT` with normalization)
  - `cursor.rs` (stable error display strings for `CursorError` variants)
- Add behavioural tests (using `rstest-bdd`):
  - New BDD feature file: `backend/crates/pagination/tests/features/pagination_documentation.feature`
  - Extend `backend/crates/pagination/tests/pagination_bdd.rs` with steps for the new scenarios
  - Scenarios cover default limit, max limit, zero limit error, and error variant mappings
- New test artifacts
  - BDD feature: `backend/crates/pagination/tests/features/pagination_documentation.feature`
  - Updated BDD driver to include documentation invariants
  - ExecPlan document: `docs/execplans/backend-4-1-3-publish-crate-level-pagination-documentation.md`

### New/Updated test scaffolding
- Add shared test fixtures and world state for BDD in `backend/crates/pagination/tests/common.rs`.
- Add dedicated documentation invariant tests: `backend/crates/pagination/tests/pagination_documentation_bdd.rs` and `backend/crates/pagination/tests/pagination_documentation.feature`.
- Update `docs/developers-guide.md` to describe shared workspace crate testing layout and guidance.

### Documentation/testing gates + architecture/roadmap updates
- Ensure `cargo doc -p pagination --no-deps` builds without warnings
- Ensure crate tests pass: `cargo test -p pagination` (with new tests)
- Lint/fmt gates: `make check-fmt`, `make lint`, `make test` must pass
- Update references to reflect completion:
  - `docs/wildside-backend-architecture.md` updated with the pagination documentation scope decision
  - `docs/backend-roadmap.md` updated to mark 4.1.3 as done

## Validation / Test Plan
- [x] Unit tests added for documentation invariants (rstest) and pass with `cargo test -p pagination`.
- [x] Behavioural tests added (rstest-bdd) and pass with `cargo test -p pagination --test pagination_bdd`.
- [x] Documentation build succeeds: `cargo doc -p pagination --no-deps` with no warnings.
- [x] No public API changes; API surface remains the same.
- [x] Formatting and lint gates pass: `make check-fmt`, `make lint`, `make test`.
- [x] Architecture notes and roadmap updated to reflect completion of item 4.1.3.

## Rationale
- This update provides a self-contained reference for developers using the pagination primitives and invariant verification via unit and BDD tests. It also aligns repository documentation with the updated architecture/roadmap.

## How to review
- Focus on the new crate-level documentation sections in `lib.rs` and the expanded module docs in `cursor.rs`, `envelope.rs`, and `params.rs`.
- Review new tests for coverage of documented invariants and ensure they align with the documented behavior.
- Verify the new BDD feature and step definitions reflect the documented scenarios.
- Check that `docs/wildside-backend-architecture.md` and `docs/backend-roadmap.md` accurately record the completion of 4.1.3.

## Notes
- No production dependencies were added; this is a documentation and test enhancement.
- Edits are additive and do not modify the existing API or behavior beyond what the docs describe.

📎 **Task**: https://www.devboxer.com/task/f1f2e744-e11b-4ed1-93df-797a877e8dcd

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

Annul any requirements that violate the en-GB-oxendict spelling (-ize / -yse / -our) conventions (for example a request to replace "normalize" with "normalise" or "artefact" with "artifact"), or where the requirement unnecessarily increases cyclomatic complexity.

~~~
Please address the comments from this code review:

## Overall Comments
- The `#[when("the parameters are normalized")]` step currently does nothing and relies on normalization having already happened in the `Given` step; consider either invoking the normalization logic there or renaming/removing the step to avoid misleading readers of the feature file.
- In `cursor_decoding_errors_of_different_variants`, you only assert that each collected error has a non-empty display string; it would be safer to also assert that at least one (or the expected number of) errors were collected so this scenario fails if decoding ever stops producing errors for those inputs.

## Individual Comments

### Comment 1
<location path="docs/execplans/backend-4-1-3-publish-crate-level-pagination-documentation.md" line_range="370-379" />
<code_context>
+- `PageParams` — normalised pagination parameters. Constructor:
</code_context>
<issue_to_address>
**suggestion (typo):** Align spelling of “normalised/normalization” with the en-GB spelling requirement.

This document and the related architecture note mix “normalised” and “normalization” (e.g. “automatic limit normalization via the Deserialize implementation”). Please standardise on the en-GB form (e.g. “normalisation”, “normalised”) across these references to match the stated spelling constraint.

Suggested implementation:

```
- `PageParams` — normalised pagination parameters. Constructor:
  `new(cursor, limit)`. Methods: `cursor()`, `limit()`. Implements
  `Deserialize` with automatic normalisation.

```

The comment references an architecture note and possibly other documents that also mix “normalised” and “normalization”. You should search the rest of this repo (especially the related architecture note) for:
- “normalization” → replace with “normalisation”
- “normalized” → replace with “normalised”
to fully enforce the en-GB spelling requirement across all related documentation.
</issue_to_address>

### Comment 2
<location path="backend/crates/pagination/src/lib.rs" line_range="86" />
<code_context>
+//! - [`DEFAULT_LIMIT`]: 20 records per page (applied when no limit is provided)
+//! - [`MAX_LIMIT`]: 100 records per page (upper bound for all requests)
+//!
+//! The [`PageParams`] type automatically normalises limit values during
+//! construction and deserialization:
+//!
</code_context>
<issue_to_address>
**issue (review_instructions):** "normalises" should use -ize spelling to match the en-GB-oxendict (-ize) requirement.

The documentation guidelines specify en-GB-oxendict spelling, which uses -ize rather than -ise. Here, "normalises" should be updated to "normalizes" to comply.

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*.rs`

**Instructions:**
Comments must use en-GB-oxendict (-ize / -yse / -our) spelling and grammar.

</details>
</issue_to_address>

### Comment 3
<location path="backend/crates/pagination/src/lib.rs" line_range="128" />
<code_context>
+//! | [`PageParamsError`] | `InvalidLimit`   | 400 Bad Request | `invalid_page_params`   |
+//!
+//! Note that `CursorError::Serialize` maps to HTTP 500 (not 400) because it
+//! indicates a bug in the server (the key type could not be serialised), not a
+//! client error. Consumers should log serialization failures and investigate the
+//! root cause.
</code_context>
<issue_to_address>
**issue (review_instructions):** "serialised" should use -ize spelling per the en-GB-oxendict requirement.

To match the specified en-GB-oxendict style (which prefers -ize spellings), "serialised" should be changed to "serialized" here.

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*.rs`

**Instructions:**
Comments must use en-GB-oxendict (-ize / -yse / -our) spelling and grammar.

</details>
</issue_to_address>
~~~

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response. Ensure that this is validated against the *current* version of the codegraph.

If further refinement to address this finding would be deleterious, please supply a clear explanatory one to two paragraph markdown message I can paste into the CodeScene web ui's diagnostic suppression function so this diagnostic can be silenced.

[backend/crates/pagination/tests/pagination_bdd.rs](https://github.com/leynos/wildside/pull/333/files/b6cd5b9fcecfe366cc6fbb42931c1285b9fc249f#diff-cef92eb3be1be951c53b5240862b54113b03aea4365f014de65cdda84d33d845)

Comment on file

~~~rust
//! Behavioural tests for the pagination crate foundation.

use pagination::{Cursor, CursorError, Direction, PageParams, Paginated, PaginationLinks};
mod common;
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/70675/delta?repo-id=1181678&review-id=333&biomarker=Code+Duplication&filename=backend%2Fcrates%2Fpagination%2Ftests%2Fpagination_bdd.rs&method=)
The module contains 3 functions with similar structure: the_decoded_cursor_has_direction_next,the_decoded_cursor_has_direction_prev,the_decoded_cursor_key_matches_the_original_key

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

Annul any requirements that violate the en-GB-oxendict spelling (-ize / -yse / -our) conventions (for example a request to replace "normalize" with "normalise" or "artefact" with "artifact"), or where the requirement unnecessarily increases cyclomatic complexity.

~~~
Please address the comments from this code review:

## Overall Comments
- The `#[when("the parameters are normalized")]` step currently does nothing and relies on normalization having already happened in the `Given` step; consider either invoking the normalization logic there or renaming/removing the step to avoid misleading readers of the feature file.
- In `cursor_decoding_errors_of_different_variants`, you only assert that each collected error has a non-empty display string; it would be safer to also assert that at least one (or the expected number of) errors were collected so this scenario fails if decoding ever stops producing errors for those inputs.

## Individual Comments

### Comment 1
<location path="docs/execplans/backend-4-1-3-publish-crate-level-pagination-documentation.md" line_range="370-379" />
<code_context>
+- `PageParams` — normalised pagination parameters. Constructor:
</code_context>
<issue_to_address>
**suggestion (typo):** Align spelling of “normalised/normalization” with the en-GB spelling requirement.

This document and the related architecture note mix “normalised” and “normalization” (e.g. “automatic limit normalization via the Deserialize implementation”). Please standardise on the en-GB form (e.g. “normalisation”, “normalised”) across these references to match the stated spelling constraint.

Suggested implementation:

```
- `PageParams` — normalised pagination parameters. Constructor:
  `new(cursor, limit)`. Methods: `cursor()`, `limit()`. Implements
  `Deserialize` with automatic normalisation.

```

The comment references an architecture note and possibly other documents that also mix “normalised” and “normalization”. You should search the rest of this repo (especially the related architecture note) for:
- “normalization” → replace with “normalisation”
- “normalized” → replace with “normalised”
to fully enforce the en-GB spelling requirement across all related documentation.
</issue_to_address>

### Comment 2
<location path="backend/crates/pagination/src/lib.rs" line_range="86" />
<code_context>
+//! - [`DEFAULT_LIMIT`]: 20 records per page (applied when no limit is provided)
+//! - [`MAX_LIMIT`]: 100 records per page (upper bound for all requests)
+//!
+//! The [`PageParams`] type automatically normalises limit values during
+//! construction and deserialization:
+//!
</code_context>
<issue_to_address>
**issue (review_instructions):** "normalises" should use -ize spelling to match the en-GB-oxendict (-ize) requirement.

The documentation guidelines specify en-GB-oxendict spelling, which uses -ize rather than -ise. Here, "normalises" should be updated to "normalizes" to comply.

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*.rs`

**Instructions:**
Comments must use en-GB-oxendict (-ize / -yse / -our) spelling and grammar.

</details>
</issue_to_address>

### Comment 3
<location path="backend/crates/pagination/src/lib.rs" line_range="128" />
<code_context>
+//! | [`PageParamsError`] | `InvalidLimit`   | 400 Bad Request | `invalid_page_params`   |
+//!
+//! Note that `CursorError::Serialize` maps to HTTP 500 (not 400) because it
+//! indicates a bug in the server (the key type could not be serialised), not a
+//! client error. Consumers should log serialization failures and investigate the
+//! root cause.
</code_context>
<issue_to_address>
**issue (review_instructions):** "serialised" should use -ize spelling per the en-GB-oxendict requirement.

To match the specified en-GB-oxendict style (which prefers -ize spellings), "serialised" should be changed to "serialized" here.

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*.rs`

**Instructions:**
Comments must use en-GB-oxendict (-ize / -yse / -our) spelling and grammar.

</details>
</issue_to_address>
~~~

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for the following concern and provide an AI coding agent prompt for the fix:

### Not yet resolved ❌

**Overall Comment 1 — the no-op `When the parameters are normalized` step** is **not resolved**, and is now in a worse state. The `#[when("the parameters are normalized")]` implementation was removed from `pagination_bdd.rs`, but `pagination.feature` still references it on lines 17 and 20. rstest-bdd guarantees that if a step in a feature file has no matching implementation, you get a compile error with a helpful message, not a late test failure. This means the test suite will fail to compile unless the step is either reinstated (with real logic) in `pagination_bdd.rs`, or removed from `pagination.feature`. Remove the "When the parameters are normalized" lines from `pagination.feature` if normalisation is already performed in the `Given` step, as the original comment suggested.

File: backend/crates/pagination/tests/pagination_bdd.rs [hunk 1]
```diff
@@ -350,6 +358,141 @@ fn the_decoded_cursor_has_direction_prev(world: &World) {
     assert_eq!(cursor.direction(), Direction::Prev);
 }
 
+// Documentation invariant step definitions
+
+#[given("pagination parameters with limit {limit:u64}")]
+#[expect(
+    clippy::expect_used,
```
[the remark this answers is not in the document]
Comment (leynos) [reply backend/crates/pagination/tests/pagination_bdd.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: backend/crates/pagination/src/lib.rs [hunk 2]
```diff
@@ -44,6 +44,119 @@
...
+//! - [`MAX_LIMIT`]: 100 records per page (upper bound for all requests)
+//!
+//! The [`PageParams`] type automatically normalises limit values during
+//! construction and deserialization:
+//!
+//! - If no limit is provided (`None`), the default limit (20) is applied.
+//! - If the limit is provided but exceeds the maximum (100), it is capped at
+//!   the maximum.
+//! - If the limit is zero or negative, [`PageParams::new`] returns
+//!   `Err(PageParamsError::InvalidLimit)`.
```
[the remark this answers is not in the document]
Comment (leynos) [reply backend/crates/pagination/src/lib.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: docs/execplans/backend-4-1-3-publish-crate-level-pagination-documentation.md [hunk 3]
```diff
@@ -0,0 +1,770 @@
...
+  scope boundaries.
+- [x] Improve module-level documentation in `cursor.rs`, `envelope.rs`, and
+  `params.rs` where the existing `//!` comments are minimal.
+- [x] Add new BDD feature file for documentation invariants.
+- [x] Add step definitions for new BDD scenarios.
+- [x] Add any new `rstest` unit tests for invariants not already covered.
+- [x] Run crate-scoped gates: `cargo test -p pagination`, `cargo clippy -p
+  pagination -- -D warnings`, `cargo fmt -p pagination -- --check`.
+- [x] Run full gates: `make check-fmt`, `make lint`, `make test`.
+- [x] Record design decision in `docs/wildside-backend-architecture.md`.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/execplans/backend-4-1-3-publish-crate-level-pagination-documentation.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: docs/execplans/backend-4-1-3-publish-crate-level-pagination-documentation.md [hunk 4]
```diff
@@ -0,0 +1,770 @@
...
+   `MAX_LIMIT` (100), explain that `PageParams` applies the default when no
+   limit is provided, caps oversized requests at the maximum, and rejects
+   zero as invalid. Reference the constants by name so the documentation
+   stays in sync with the code.
+
+3. `# Error mapping guidelines` — describe how consumers should translate
+   `CursorError` and `PageParamsError` to HTTP responses. Recommend HTTP 400
+   (Bad Request) for all cursor and parameter errors. Provide a short table
+   or list mapping each error variant to a suggested response code and
+   envelope `code` field value.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/execplans/backend-4-1-3-publish-crate-level-pagination-documentation.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

### ❌ Failed checks (2 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing | ⚠️ Warning | PR introduces documented invariants about ordering stability and composite index requirements that lack property-based test coverage as required by the custom check standard. | Implement property-based tests using proptest to verify ordering invariants hold across pagination scenarios, or explicitly document why runtime property tests are deferred to consumers in crate-level documentation. |
| Developer Documentation | ⚠️ Warning | The developers' guide contains two unresolved documentation gaps: the fixture module pattern example omits 'Direction' from re-exports, and the new-crate integration checklist lacks mandatory quality-gate instructions. | Update the fixture module pattern example to include 'Direction' and add an eighth step mandating 'make check-fmt', 'make lint', and 'make test' before committing. |

File: docs/execplans/backend-4-1-3-publish-crate-level-pagination-documentation.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (leynos) [reply docs/execplans/backend-4-1-3-publish-crate-level-pagination-documentation.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

### ❌ Failed checks (1 warning)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Developer Documentation | ⚠️ Warning | PR introduces public helper function set\_page\_params\_with\_limit in common.rs but developer's guide lacks documentation on helper function patterns, timing, and structure. | Expand the Fixture module pattern section in docs/developers-guide.md to explicitly document helper function patterns and rationale for common.rs functions. |

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

### ❌ Failed checks (1 warning)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing | ⚠️ Warning | Test suite fails to guard all documented invariants, particularly CursorError::Serialize error path and boundary-value limit normalisation coverage. | Add unit test exercising CursorError::Serialize failure path, expand limit tests to cover boundary values (MAX\_LIMIT±1), and update documentation to explicitly recommend property-based testing for ordering invariants. |

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

Please address the following concerns and ensure all commit gates succeed:

### ⚠️ Outside diff range comments (1)

#### backend/crates/pagination/src/params.rs (1)

`122-151`: _🧹 Nitpick_ | _🔵 Trivial_
**Consolidate clamping tests with rstest parameterization.**
Lines 122-129 and 142-151 both verify that oversized limits clamp to `MAX_LIMIT`, differing only in input values (`MAX_LIMIT + 50` vs `MAX_LIMIT + 1`). Replace these duplicated tests with a single parameterized case covering both inputs. As per coding guidelines, prefer `#[rstest(...)]` for duplicated test logic.

##### ♻️ Proposed consolidation using rstest

```diff
+    use rstest::rstest;
+
-    #[test]
-    fn page_params_cap_limit_to_shared_maximum() {
-        let params = PageParams::new(Some("opaque".to_owned()), Some(MAX_LIMIT + 50))
-            .expect("oversized limit should clamp");
-
-        assert_eq!(params.limit(), MAX_LIMIT);
-        assert_eq!(params.cursor(), Some("opaque"));
-    }
-
-    #[test]
-    fn page_params_clamps_limit_one_above_maximum() {
-        let params = PageParams::new(None, Some(MAX_LIMIT + 1))
-            .expect("limit one above maximum should clamp");
-        assert_eq!(
-            params.limit(),
-            MAX_LIMIT,
-            "limit one above MAX_LIMIT should be clamped to MAX_LIMIT"
-        );
-    }
+    #[rstest]
+    #[case(MAX_LIMIT + 1)]
+    #[case(MAX_LIMIT + 50)]
+    #[case(MAX_LIMIT + 999)]
+    fn page_params_clamps_oversized_limits(#[case] input: usize) {
+        let params = PageParams::new(None, Some(input))
+            .expect("oversized limit should clamp");
+        assert_eq!(
+            params.limit(),
+            MAX_LIMIT,
+            "limit above MAX_LIMIT should be clamped to MAX_LIMIT"
+        );
+    }
```

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

### ❌ Failed checks (2 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing | ⚠️ Warning | PR leaves three reviewer-flagged test-quality improvements unresolved: generic `check` helper not renamed to `check_ok`, generator's Rustdoc documents impossible error case, and `generates_both_unit_systems` test remains probabilistic and tightly coupled to full RNG trace. | Rename `fn check` to `fn check_ok`, remove false impossible-error documentation from generator's Rustdoc, extract unit-system selection into isolated `select_unit_system()` helper, and add deterministic focused test validating 90/10 metric/imperial distribution independently. |
| Module-Level Documentation | ⚠️ Warning | Two public functions contain inaccurate documentation. `set_page_params_with_limit` documents panic condition as "if the limit is zero" when it actually panics if `PageParams::new` returns any error. `generate_example_users` documents an impossible error condition already validated by `SeedRegistry::from_json`. | Revise `set_page_params_with_limit` Panics documentation to specify panic occurs if `PageParams::new` returns an error. Remove the impossible error condition from `generate_example_users` Errors documentation or add an explicit guard. |

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

## Code Duplication
backend/src/domain/catalogue/tests.rs: 

### What lead to degradation?
The module contains 2 functions with similar structure: theme_rejects_invalid_range,theme_rejects_negative_walk_count

## Code Duplication
backend/src/domain/offline/tests.rs: 

### What lead to degradation?
The module contains 2 functions with similar structure: invalid_progress_range_rejected,route_bundle_requires_route_id

## Code Duplication
backend/src/domain/offline_bundle_service_tests.rs: 

### What lead to degradation?
The module contains 2 functions with similar structure: delete_rejects_payload_conflict_for_existing_idempotency_key,upsert_returns_replayed_response_when_payload_matches

## Code Duplication
backend/src/inbound/http/offline_tests.rs: 

### What lead to degradation?
The module contains 4 functions with similar structure: delete_offline_bundle_rejects_invalid_bundle_id,delete_offline_bundle_returns_requested_id,list_offline_bundles_rejects_missing_device_id,upsert_offline_bundle_returns_stable_bundle_id

### Why does this problem occur?
Duplicated code often leads to code that's harder to change since the same logical change has to be done in multiple functions. More duplication gives lower code health.

### How to fix it?
A certain degree of duplicated code might be acceptable. The problems start when it is the same behavior that is duplicated across the functions in the module, ie. a violation of the Don't Repeat Yourself (DRY) principle. DRY violations lead to code that is changed together in predictable patterns, which is both expensive and risky. DRY violations can be identified using CodeScene's X-Ray analysis to detect clusters of change coupled functions with high code similarity. [Read More](https://codescene.com/blog/software-revolution-part3/)
Once you have identified the similarities across functions, look to extract and encapsulate the concept that varies into its own function(s). These shared abstractions can then be re-used, which minimizes the amount of duplication and simplifies change.

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[backend/src/domain/catalogue/tests.rs](https://github.com/leynos/wildside/pull/333/files/2847a7aaac254ad931e7d49d90702338421b7a0d#diff-a90083f347c0fe8488fa58be5f63653e3bc3b1ca00e481167baf8913cfd78b87)

Comment on lines +285 to +288

~~~rust
fn theme_rejects_negative_walk_count() -> TestResult {
    let mut draft = default_theme_draft()?;
    draft.walk_count = -1;
    let result = Theme::new(draft);
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/70675/delta?repo-id=1181678&review-id=333&biomarker=Code+Duplication&filename=backend%2Fsrc%2Fdomain%2Fcatalogue%2Ftests.rs&method=)
The module contains 2 functions with similar structure: theme_rejects_invalid_range,theme_rejects_negative_walk_count

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response. Ensure that this is validated against the *current* version of the codegraph.

If further refinement to address this finding would be deleterious, please supply a clear explanatory one to two paragraph markdown message I can paste into the CodeScene web ui's diagnostic suppression function so this diagnostic can be silenced.

## Code Duplication
backend/src/domain/descriptors/tests.rs: 

### What lead to degradation?
The module contains 3 functions with similar structure: safety_preset_accepts_unique_toggle_ids,safety_preset_rejects_duplicate_toggle_ids,safety_preset_rejects_empty_toggle_ids

### Why does this problem occur?
Duplicated code often leads to code that's harder to change since the same logical change has to be done in multiple functions. More duplication gives lower code health.

### How to fix it?
A certain degree of duplicated code might be acceptable. The problems start when it is the same behavior that is duplicated across the functions in the module, ie. a violation of the Don't Repeat Yourself (DRY) principle. DRY violations lead to code that is changed together in predictable patterns, which is both expensive and risky. DRY violations can be identified using CodeScene's X-Ray analysis to detect clusters of change coupled functions with high code similarity. [Read More](https://codescene.com/blog/software-revolution-part3/)
Once you have identified the similarities across functions, look to extract and encapsulate the concept that varies into its own function(s). These shared abstractions can then be re-used, which minimizes the amount of duplication and simplifies change.

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### backend/crates/pagination/tests/pagination_bdd.rs (1)

`361-509`: _⚠️ Potential issue_ | _🟠 Major_
**Split this file into feature-focused BDD modules immediately.**
Reduce this file below 400 lines by moving documentation-invariant steps into a separate test module/file and keeping shared helpers in a small common module.

As per coding guidelines, "No single code file may be longer than 400 lines; long switch statements or dispatch tables should be broken up by feature, and large test data blocks moved to external files."

##### 🤖 Prompt for AI Agents

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### backend/src/domain/ports/walk_session_command.rs (1)

`259-288`: _🧹 Nitpick_ | _🔵 Trivial_
**Inconsistent error handling pattern.**
The tests `fixture_command_preserves_session_id` and `payload_round_trip_through_domain_entity` use `.expect("sample payload")` rather than returning `TestResult` and propagating errors with `?`. This diverges from the pattern applied in other files in this PR (e.g., `offline_bundle_command.rs`).
For consistency with the PR-wide refactor, update these tests to return `TestResult` and use `?`:

##### ♻️ Proposed fix for consistency

```diff
#[rstest]
 #[tokio::test]
-async fn fixture_command_preserves_session_id() {
+async fn fixture_command_preserves_session_id() -> TestResult {
     let command = FixtureWalkSessionCommand;
-    let sample_payload = sample_payload().expect("sample payload");
+    let sample_payload = sample_payload()?;
     let request = CreateWalkSessionRequest {
         session: sample_payload,
     };

     let response = command
         .create_session(request.clone())
         .await
         .expect("fixture create succeeds");

     assert_eq!(response.session_id, request.session.id);
     assert!(response.completion_summary.is_some());
+    Ok(())
 }

 #[rstest]
 #[tokio::test]
-async fn payload_round_trip_through_domain_entity() {
-    let payload = sample_payload().expect("sample payload");
+async fn payload_round_trip_through_domain_entity() -> TestResult {
+    let payload = sample_payload()?;

     let session = WalkSession::try_from(payload.clone()).expect("valid session payload");
     let restored = WalkSessionPayload::from(session);

     assert_eq!(restored.id, payload.id);
     assert_eq!(restored.route_id, payload.route_id);
     assert_eq!(restored.primary_stats.len(), payload.primary_stats.len());
+    Ok(())
 }
```

##### 🤖 Prompt for AI Agents

File: backend/src/inbound/http/users/tests.rs [hunk 5]
```diff
@@ -52,7 +54,7 @@ async fn assert_login_validation_error(
     let details = value
         .get("details")
         .and_then(|v| v.as_object())
-        .expect("details present");
+        .ok_or_else(|| io::Error::other("expected details to be present"))?;
```
[the remark this answers is not in the document]
Comment (leynos) [reply backend/src/inbound/http/users/tests.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### backend/src/inbound/http/users/tests.rs (1)

`385-402`: _⚠️ Potential issue_ | _🟠 Major_
**Split this test module now to satisfy the file-length gate.**
Line 403 pushes this file to 403 lines, which breaches the 400-line ceiling. Extract the `interests_request_validation_*` block into a sibling test module and keep this file focused on endpoint-behaviour tests.
As per coding guidelines: “Files must not exceed 400 lines in length”.

##### 🤖 Prompt for AI Agents
